sklearn/utils/_testing.py

```diff
@@ -82,9 +82,18 @@
 assert_raises_regexp = assert_raises_regex
 
 
+# TODO: Remove in 1.2
+@deprecated(  # type: ignore
+    "`assert_warns` is deprecated in 1.0 and will be removed in 1.2."
+    "Use `pytest.warns` instead."
+)
 def assert_warns(warning_class, func, *args, **kw):
     """Test that a certain warning occurs.
 
+    .. deprecated:: 1.0
+        assert_warns is deprecated in 1.0 and will be removed in 1.2.
+        Use pytest.warns instead.
```

Comment:
```suggestion
        `assert_warns` is deprecated in 1.0 and will be removed in 1.2.
        Use :func:`pytest.warns` instead.
```

I don't know if we have added `pytest` to get an hyperlink, thought.

Reply:
As explained in the description of the PR, it's not really public API so it's not rendered on our API website. As a result, I would be in favor of not committing this suggestion.

Reply:
True.